### ISSUE TITLE
Ability to add HTTP request header for calls to the WebServiceTarget

### ISSUE THREAD

**Issue by devJ0n** [NLog/NLog#1996]
**Type**:
Feature request

**NLog version**:
Latest

**Platform**:
.Net 3.5 / .Net 4.0 / .Net 4.5 / Mono 4 / Silverlight 4 / Silverlight 5 / Xamarin Android / Xamarin iOs / .NET Core

**Current NLog config** (xml or C#, if relevant)
```xml
<nlog>
...
<targets>
  <!-- Write logs to webservice -->
    <target name="logMessageWebServiceBuffered" xsi:type="BufferingWrapper" slidingTimeout="true" bufferSize="100" flushTimeout="60000">
      <target xsi:type="WebService" url="https://acme.com/api/log" protocol="JsonPost" encoding="UTF-8" includeBOM="false">
        <parameter name="Message"    layout="${message}" />
        <parameter name="DateTime"    layout="${event-properties:item=DateTime}" />
        <parameter name="SystemName" layout="${event-properties:item=SystemName}" />
        <parameter name="ApplicationName" layout="${event-properties:item=ApplicationName}" />
        <parameter name="JobName" layout="${event-properties:item=JobName}" />
        <parameter name="Status" layout="${event-properties:item=Status}" />
      </target>
    </target>
</targets>
...
</nlog>
```
 
 - Why do we need it?
Currently logging towards an webservice (REST API) target using the JsonPost format is unable to add HTTP request headers. An example might be that the webservice is protected and require a bearer token in the authorization header of the HTTP request in order to use the webservice.

A HTTP request for the JsonPost protocol could look like this:
```
POST /api/log HTTP/1.1
Host: acme.com
Content-Type: application/json
Authorization: Bearer XVk2kfc-cmkasdjiandasdnmlasdmklifcnmasdmklasd223123123zzxdzxxczxcz-asr34dfsdfsdf

[
  {
    'message': 'Order was created',
    'dateTime': 2017-02-08T18:25:42.437Z,
    'systemName': 'OrderSystem',
    'applicationName': 'AspNet MVC WebOrderClient',
    'jobName': 'OrderCreated',
    'status': 83
  },
  {
    'message': 'Order was confirmed',
    'dateTime': 2017-02-08T18:25:43.511Z,
    'systemName': 'OrderSystem',
    'applicationName': 'AspNet MVC WebOrderClient',
    'jobName': 'OrderConfirmation',
    'status': 84
  },
  ...
]
```

I have no good knowledge about NLog and how to implement this according to the current conventions but I'm open for discussions and believe this is a important future feature.

**Comment by 304NotModified** [comment]
This should be possible already by using the layout renderers of NLog.web, see https://github.com/NLog/NLog/wiki/Layout-Renderers#nlogweb-package-

**Comment by mohyounis** [comment]
Hi,
 Do you have an example on how to pass the authorization with WebService target?

**Comment by 304NotModified** [comment]
have to search for it. Could you try on stackoverflow?

**Comment by mohyounis** [comment]
Please let me know if you find. I'm new to NLOG and what I'm trying to do is be able to use WebService target (httppost protocol) and allow the use of request header request.header["Authorization"]. Appreciate in advance your assiatnce

**Comment by snakefoot** [comment]
Added support for HttpWebRequest Headers in #1912. Not sure if it should be pushed to NLog 5.0, as it is a breaking change.

**Comment by snakefoot** [comment]
@304NotModified Resolved with #1912

**Comment by 304NotModified** [comment]
thanks @snakefoot !
